File: Code_Help_By_Babbar/23_2D_arrays/05_largest_row_sum.cpp
#include <iostream>
using namespace std;

int main()
{
    int arr[3][4] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12};

    int maxSum = INT_MIN;
    int rowIdx = -1;

    for (int i = 0; i < 3; i++)
    {
        int sum = 0;
        for (int j = 0; j < 4; j++)
        {
            sum = sum + arr[i][j];
            if (sum > maxSum)
            {
                maxSum = sum;
                rowIdx = i;
            }
        }
    }
    cout << "The largest row sum is : " << maxSum << endl;
    cout << "The largest row index is : " << rowIdx << " ";

    return 0;
}

File: Topics_to_Focus_On_JS/[ 01 ] Basic Data Structure/01_Array/03_Matrix/05_largest_row_sum.js
function largeRowSum(arr, row, col) {
  const matrix = Array.from({ length: row }, () => new Array(col).fill(0));
  let idx = 0;
  let maxSum = -Infinity;

  // print row wise
  for (let m = 0; m < row; m++) {
    for (let n = 0; n < col; n++) {
      matrix[m][n] = arr[idx++];
    }
  }

  // row wise sum
  for (let m = 0; m < row; m++) {
    let sum = 0;
    for (let n = 0; n < col; n++) {
      sum = sum + matrix[m][n];
      if (sum > maxSum) {
        maxSum = sum;
      }
    }
    console.log(sum);
  }

  return maxSum;
}

const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const row = 3;
const col = 3;
const res = largeRowSum(arr, row, col);
console.log(res);
